tests/client/test_base_client.py

```diff
@@ -811,3 +831,22 @@ def test_client__json_decode_error(dynamics_client):
 
     with pytest.raises(DynamicsException):
         dynamics_client.delete()
+
+
+def test_client__connection_timeout():
```

Comment:
I think this is enough of a test for this, though it doesn't test what happens when the timeout happens.